File: domain/src/main/java/com/hanghae/domain/entity/Showtime.java
package com.hanghae.domain.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDateTime;

@Entity
@Getter
@Setter
@NoArgsConstructor
public class Showtime extends BaseEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer showtimeId; // 상영 일정 ID

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "movie_id", nullable = false)
    private Movie movie; // 영화 (외래키)

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "theater_id", nullable = false)
    private Theater theater; // 상영관 (외래키)

    private LocalDateTime startTime; // 상영 시작 시간

    private LocalDateTime endTime; // 상영 종료 시간
}

File: domain/src/main/kotlin/entity/ShowTime.kt
package entity

import java.time.LocalDateTime
import java.util.*

data class ShowTime(
    val id: UUID,                   // 고유 ID
    val movieId: UUID,              // 영화 고유 ID
    val screenId: UUID,             // 스크린 고유 ID
    val startTime: LocalDateTime    // 시작 시간
)
